applications/callflow/src/module/cf_voicemail.erl

```diff
@@ -1366,7 +1367,8 @@ get_caller_id_number(Call) ->
     CallerIdNumber = whapps_call:caller_id_number(Call),
     case whapps_call:kvs_fetch('prepend_cid_number', Call) of
         'undefined' -> CallerIdNumber;
-        Prepend -> <<(wh_util:to_binary(Prepend))/binary, CallerIdNumber/binary>>
+        Prepend -> Pre = <<(wh_util:to_binary(Prepend))/binary, CallerIdNumber/binary>>,
+                   wh_util:truncate_right_binary(Pre, 35)
```

Comment:
We could and I did it in a local function first, but I thought it should be nice to have it as a generic function so other modules can using it if they need it.

From: Pierre Fenoll <notifications@github.com<mailto:notifications@github.com>>
Reply-To: 2600hz/kazoo <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Thursday, December 3, 2015 at 12:41 AM
To: 2600hz/kazoo <kazoo@noreply.github.com<mailto:kazoo@noreply.github.com>>
Cc: Hesaam Farhang <hesaam@2600hz.com<mailto:hesaam@2600hz.com>>
Subject: Re: [kazoo] KAZOO-4321: Truncate prepend caller id to not exceed characters long limit (#1465)


In applications/callflow/src/module/cf_voicemail.erl<https://github.com/2600hz/kazoo/pull/1465#discussion_r46525010>:

> @@ -1366,7 +1367,8 @@ get_caller_id_number(Call) ->
>      CallerIdNumber = whapps_call:caller_id_number(Call),
>      case whapps_call:kvs_fetch('prepend_cid_number', Call) of
>          'undefined' -> CallerIdNumber;
> -        Prepend -> <<(wh_util:to_binary(Prepend))/binary, CallerIdNumber/binary>>
> +        Prepend -> Pre = <<(wh_util:to_binary(Prepend))/binary, CallerIdNumber/binary>>,
> +                   wh_util:truncate_right_binary(Pre, 35)


Yeah how about just a local private function truncate_right/1 which would just be your wh_util:truncate_right_binary/2? Let's not add more useless stuff to wh_util…

—
Reply to this email directly or view it on GitHub<https://github.com/2600hz/kazoo/pull/1465/files#r46525010>.